HHS/Head-Start-TTADP
# Recipient record filter validation, UI fixes

## Description of change
- Filters should validate on blur. Depending on which field is missing
- Clicking the individual apply buttons are not necessary
- The date picker doesn't blur when you click within it
- The date input has a border

Additionally, a bug with data reloading was fixed (useDeepCompareEffect seemed to be the culprit - to see what I mean, go to prod, change the query in the date filter without adding or removing a new filter. I observed that it wasn't reloading the data, which I also ran into when adding the onChange handlers to the dropdown components). Fixing that required a little bit of extra poking around in the codebase.

Not attempting to solve every UI issue with the date picker here - theres a ticket that needs refinin' that goes deeper.

## How to test
- The filter validation works as expected
- The blur behavior in general makes sense
- You can update the roles and the date without clicking the "secondary" apply buttons, just the big fella at the bottom
- API should not be called excessively

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-500

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
